sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/models/BlobDownloadResponse.java

```diff
@@ -12,21 +12,21 @@
 /**
  * This class contains the response information returned from the server when downloading a blob.
  */
-public final class BlobDownloadAsyncResponse extends ResponseBase<BlobDownloadHeaders, ResponseBody> {
+public final class BlobDownloadResponse extends ResponseBase<BlobDownloadHeaders, ResponseBody> {
```

Comment:
Java has two distinct types: [`BlobDownloadResponse`](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobDownloadResponse.java) and [`BlobDownloadAsyncResponse`](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobDownloadAsyncResponse.java). I originally decided to go for the latter because our use case lined up with what Java has.

Do you think we will add something like Java's `BlobDownloadResponse` where no ResponseBody is returned but rather written to something like an `OutputStream`? I tend to think so.

Reply:
Right, Java has `BlobDownloadAsyncResponse` to support `Flux` that asynchronously streams the bytes from the connection, so the text "async" in the model indicates that it composes "async" Flux.

In the Android case, `BlobDownloadAsyncResponse`  is composing `ResponseBody` that exposes `InputStream` from which we can "synch"-ronously read the bytes from the connection. This is why I removed the text, in addition to that, it looks also weird when that type listed in the JavaDoc with our other Response types.

Reply:
Since the decision to read the bytes synchronously would come down to us, do you think there could be a point where we would want to go the fully async route with `ResponseBody`? (i.e. using [`byteStream`](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-response-body/byte-stream/)).

Reply:
`byteStream()` API returns "InputStream", in my previous comment when I said "sync" read I mean that the InputStream::read API itself is blocking. i.e. can't do something like below with IS:

```java
InputStream.readAsync(bytes_count, new CallBack(byte[] bytes) {
});
```

So even though we can asynchronously return `BlobDownloadResponse`, we can't asynchronously read the content:

```java
downloadAsync(params, new CallBack<BlobDownloadResponse>() {
    // This is async
    public void onResponse(BlobDownloadResponse res) {
         // below read is not async but sync.
         res.resposeBody.byteStream().read();
    }
}
```

Reply:
Does this mean we won't be able to asynchronously read content if we are using OkHttp?

Reply:
yes, OkHttp is natively built on blocking socket not based on selectable channel(e.g. netty). But it should be fine in the mobile space, even if we use native UrlConnection (like android SDK from other cloud providers), it still sync. I think we don't have to worry about it at least now.

Reply:
Thanks for the thorough explanation! :)